tests/test-cases/Group1-Docker-Commands/1-02-Docker-Pull.robot

```diff
@@ -16,9 +16,85 @@
 Documentation  Test 1-02 - Docker Pull
 Resource  ../../resources/Util.robot
 Suite Setup  Conditional Install VIC Appliance To Test Server
-Suite Teardown  Cleanup VIC Appliance On Test Server
+Suite Teardown  Local Teardown
 Test Timeout  20 minutes
 
+*** Keywords ***
+Local Teardown
+    Run Keyword  Cleanup VIC Appliance On Test Server
+    Run Keyword  Cleanup VCH Bridge Network  %{BRIDGE_NETWORK_2}
+
+Get And Run MITMProxy Container
+    # Need to change this container? Read README.md in vic/tests/resources/dockerfiles/docker-pull-mitm-proxy
+    Wait Until Keyword Succeeds  5x  15 seconds  Pull image  victest/docker-layer-injection-proxy:latest
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} run --network public -itd --name=mitm -p 8080:8080 victest/docker-layer-injection-proxy
+    Log  ${output}
+    Should Be Equal As Integers  ${rc}  0
+
+Get And Run Prepared Registry
+    # Need to change this container? Read README.md in vic/tests/resources/dockerfiles/docker-pull-mitm-proxy
+    Wait Until Keyword Succeeds  5x  15 seconds  Pull image  victest/registry-busybox:latest
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} run -itd --network public --name=registry -p 5000:5000 victest/registry-busybox
+    Log  ${output}
+    Should Be Equal As Integers  ${rc}  0
+
+Run Docker Ps
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} ps
+    Should Be Equal As Integers  ${rc}  0
+    [Return]  ${output}
+
+Get Container Address
+    [Arguments]  ${container-name}  ${docker-ps}
+    ${rc}  ${container}=  Run And Return Rc And Output  echo '${docker-ps}' | grep ${container-name} | awk '{print $(NF-1)}' | cut -d- -f1
+    Should Be Equal As Integers  ${rc}  0
+    Log  ${container}
+    [Return]  ${container}
+
+Pull And MITM Prepared Image
+    [Arguments]  ${vch2-params}  ${registry}
+    ${rc}  ${output}=  Run And Return Rc And Output  docker ${vch2-params} pull ${registry}/busybox
+    Log  ${output}
```

Comment:
is this complete?  Seems cut-off in the middle?

Reply:
or at least not doing what the name of the keyword implies it is intended to do..

Reply:
It does do what the keyword implies it is intended to do. VCH-2 is deployed to use MITMproxy as a proxy and the prepared registry. The `pull` exercises the bug, just like in the bug report, and in the real world.